pkg/signatures/cosign_sig_verifier.go

```diff
@@ -396,7 +396,11 @@ func verifyImageSignature(ctx context.Context, signature oci.Signature,
 	if !bundleVerified && !cosignOpts.IgnoreTlog {
 		return nil, errUnverifiedBundle
 	}
-	return getVerifiedImageReference(signature, image)
+	refs, err := getVerifiedImageReference(signature, image)
+	if len(refs) == 0 && err == nil {
+		return nil, errors.New("no verified references")
+	}
+	return refs, err
```

Comment:
I would first check the error, if not-nil, then wrap & return; next check `len` and return error.

It is okay the way it is - I trust you know the logic here.